server/drivers/crate/index.js

```diff
@@ -69,6 +67,26 @@ async function runQuery(query, connection) {
   }
 }
 
+/**
+ * Get connection string with options
+ * @param {*} connection
+ */
+function getConnectionString(connection) {
+  let connectionString = connection.ssl ? 'https://' : 'http://';
+  if (connection.username !== undefined && connection.username.length !== 0) {
```

Comment:
I'm not sure if it is possible, but `null` could potentially be passed in and would break without checking for it. 

Can this check be changed to either `if (connection.username)`, or `if (typeof connection.username === 'string' && connection.username.length !== 0)`?

`null`, `undefined` and empty string`''` all evaluate as falsy

Reply:
I changed it to `if (connection.username)`